spyder_kernels/customize/spydercustomize.py

```diff
@@ -674,6 +677,49 @@ def runfile(filename, args=None, wdir=None, namespace=None, post_mortem=False):
 builtins.runfile = runfile
 
 
+def runcell(cell_name, filename):
+    """
+    Runs a code cell from an editor as a file.
+
+    Currently looks for code in in a ipython property called `cell_code`.
+    This property must be set by the editor prior to calling this function
+    This function deletes the contents of `cell_code` upon completion.
+
+    The filename is needed to allow for proper traceback links.
+
+    The cell_name is used currently as a reference in the history log of which
+    cell was run with the command. It is currently unused in the function.
+
+    Parameters
+    ----------
+
+    cell_name: The name of the code cell (string)
+    filename: The name of the file containing the code cell (string)
+    """
+    try:
+        filename = filename.decode('utf-8')
+    except (UnicodeError, TypeError, AttributeError):
+        # UnicodeError, TypeError --> eventually raised in Python 2
+        # AttributeError --> systematically raised in Python 3
+        pass
+    run_umr()
+    namespace = _get_globals()
+    namespace['__file__'] = filename
+    ipython_shell = get_ipython()
+    try:
+        cell_code = ipython_shell.cell_code
+    except AttributeError:
+        _print("--Run Cell Error--\n"
+               "Please use only in editor")
```

Comment:
For clarity, you could say something like ``Please use only through Spyder's Editor; shouldn't be called manually.``